File: DE101 portfolio Python files/election.py
class Election:
    def __init__(self, new_year):
        self.year = new_year
        self.all_parties = []
        self.all_electorates = []
        self.party_count = 0
        self.electorate_count = 0

    def __str__(self):
        result = f'The Election {self.year} \n'
        return result

    def add_party(self, new_name):
        a_party = Party(new_name)
        self.all_parties.append(a_party)
        self.party_count += 1

    def add_electorate(self, new_name):
        a_electorate = Electorate(new_name)
        self.all_electorates.append(a_electorate)
        self.electorate_count += 1

    def to_table(self):
        result = f'<table><tr><th colspan="3">The Election {self.year} </th></tr><tr><th>Parties</th></tr>'
        for a_party in self.all_parties:
            result += a_party.to_table_row()
        result += '<tr><th>Electorates</th></tr>'
        for a_electorate in self.all_electorates:
            result += a_electorate.to_table_row()
        result += '</table>'
        return result


class Electorate:
    def __init__(self, new_name):
        self.name = new_name

    def __str__(self):
        result = f'{self.name}'
        return result

    def to_table_row(self):
        result = '<tr>'
        result += f'<td>{self.name}</td>'
        result += '</tr>'
        return result


class Party:
    def __init__(self, new_name):
        self.name = new_name

    def __str__(self):
        result = f'{self.name}'
        return result

    def to_table_row(self):
        result = '<tr>'
        result += f'<td>{self.name}</td>'
        result += '</tr>'
        return result


if __name__ == '__main__':
    the_election = Election(2020)
    print(the_election)

File: Javascript/src/Election.js
class Election { // eslint-disable-line no-unused-vars
  constructor (newYear) {
      this.year = newYear
      this.allTheParties = []
      this.allTheElectorates = []
	  this.allTheCandidates = []
    }

    addParty(newName) {
        const newParty = new Party(newName)
        this.allTheParties.push(newParty)
        this.partyCount += 1
    }

    addElectorate(newName) {
        const newElectorate = new Electorate(newName)
        this.allTheElectorates.push(newElectorate)
        this.ElectorateCount += 1
    }

   
	findElectorate (targetElectorateName) {
		return this.allTheElectorates.find(Electorate => Electorate.name === targetElectorateName)
	}

	addCandidate(newName, aCandidateName) {
        const newCandidate = new Candidate(newName, theElectorate)
        this.allTheCandidates.push(newCandidate)
        this.candidateCount += 1
    }
	
   /*
    findParty (targetPartyName) {
        return this.allTheParties.find(aParty => aParty.name === targetPartyName)
    }


    setWinningParty (targetElectorateName, partyName ) {
    // find the electorate
    // find the party
    // tell the party about the electorate
    // tell the electorate about the party
    }


    setVotePercent(targetPartyName, newVotePercent){
    // find the party
    // pass the vote to the party
    }



	/*toString () {
		let result = `New Zealand Election ${this.year}`
			return result
	}

  /*
	sortElection () {
		this.Election.sort(function (a, b) {
			if (a.name < b.name) {
				return -1
			}
			if (a.name > b.name) {
				return 1
			} // a must be equal to b
			return 0
		} )
	} */


    toTable() {
        let result = `<table><tr><th colspan="3">THE NEW ZEALAND ELECTION ${this.year}</th></tr>`
		result += `<tr><th>PARTIES</th></tr>`
        for (const aParty of this.allTheParties) {
        result += aParty.toTableRow()
        }
		result += `<tr><th>ELECTORATES</th></tr>`
        for (const aElectorate of this.allTheElectorates) {
            result += aElectorate.toTableRow()
        }
        result += '</table>'
        return result
	}
}